modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/bulk/selection/action/EditCategoriesBulkSelectionAction.java

```diff
@@ -97,14 +98,21 @@ public void execute(
 			});
 	}
 
-	private long[] _getLongArray(Map<String, Serializable> map, String key) {
-		return ArrayUtil.toArray((Long[])map.getOrDefault(key, new Long[0]));
-	}
-
 	private Set<Long> _toLongSet(Map<String, Serializable> map, String key) {
 		try {
-			return SetUtil.fromArray(
-				(Long[])map.getOrDefault(key, new Long[0]));
+			Serializable values = map.get(key);
+
+			if (values instanceof Integer[]) {
```

Comment:
Can you first check for `Long[]` instead?
```
if (values instance of Long[]) {
    return SetUtil.fromArray(...);
}

...
```
This will make the code flatter and easier to read.

Reply:
Please see the latest changes.